pkg/operator/deploy/staticresources/worker/machinehealthcheck.yaml

```diff
@@ -0,0 +1,18 @@
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineHealthCheck
+metadata:
+  name: openshift-machinehealthcheck 
+  namespace: openshift-machine-api
+spec:
+  selector:
+    matchLabels:
+      machine.openshift.io/cluster-api-machine-role: worker 
+      machine.openshift.io/cluster-api-machine-type: worker
```

Comment:
My gut feeling is if a customer is creating custom machine sets and wanted mhc they could enable it themselves.

Having it enabled on custom machine sets by default might not be desired behaviour from a customers perspective. With a controller / operator deployment it might also be painful to deal with.

Let's gather more input from the team but I could see this being an item we pocket for now but could roll out in the future.

Reply:
I think that maybe we should have it template it out for each machine set type? Then infra nodes can have their own maxUnhealthy if they decide to deploy them, rather than having it be unable to do anything if a worker and an infra node go down.